src/components/atoms/ControlledFields/RadioButtonWithLabel/RadioButtonWithLabel.tsx

```diff
@@ -0,0 +1,40 @@
+import { FormControlLabel, Radio } from '@mui/material';
+import { useController } from 'react-hook-form';
+
+import { CheckedRadioIcon, UncheckedRadioIcon } from '~/components/atoms';
+
+import { IProps } from './types';
+
+export const RadioButtonWithLabel = <T extends object>({
+  name,
+  control,
+  label,
+  onChange,
+  ...props
+}: IProps<T>) => {
+  const { field } = useController({
+    name,
+    control,
+  });
+
+  return (
+    <FormControlLabel
+      label={label}
+      control={
+        <Radio
+          {...field}
+          checked={field.value}
+          onChange={({ target: { value: fieldValue } }) => {
+            field.onChange(fieldValue);
+            if (onChange) {
+              onChange(fieldValue);
```

Comment:
the same question about duplicating `onChange` method

Reply:
replied above. we can get rid of it and use `getValues` for getting form values and write the logic based on the form inputs values

Reply:
Good idea👍